### ISSUE TITLE
[docs][question] what is valid for linkcheck_anchors_ignore?

### ISSUE THREAD

**Issue by svenevs** [sphinx-doc/sphinx#4886]
Subject: unable to wield `linkcheck_anchors_ignore`, likely user error.

### Problem

From the [description of `linkcheck_anchors_ignore`](http://www.sphinx-doc.org/en/master/config.html#confval-linkcheck_anchors_ignore):

> A list of regular expressions that match URIs that should skip checking the validity of anchors in links. This allows skipping entire sites, where anchors are used to control dynamic pages, or just specific anchors within a page, where javascript is used to add anchors dynamically, or use the fragment as part of to trigger an internal REST request. Default is `["/#!"]`.

An example of this would be linking to an anchor on a GitHub README.  One of the links I am working with: [https://github.com/jonmiles/bootstrap-treeview#onhovercolor](https://github.com/jonmiles/bootstrap-treeview#onhovercolor).

So I thought maybe something like

```py
linkcheck_anchors_ignore = [
    r'https://github.com/jonmiles/bootstrap-treeview'
]
```

would work, or `.*bootstrap-treeview.*`, or many others.  But I cannot seem to get a regular expression that will work, and instead need to enumerate all of the anchors in `linkcheck_ignore` explicitly.

#### Procedure to reproduce the problem

With an `index.rst` and fresh sphinx project

```rst
Welcome
=======

See `this page <https://github.com/jonmiles/bootstrap-treeview#onhovercolor>`_
for more information.
```

`make linkcheck` will (correctly) report an invalid anchor.  But no `linkcheck_anchors_ignore` values I have tried in `conf.py` work.

#### Expected results

I feel like `[r'https://github.com/jonmiles/bootstrap-treeview#*']` should be accepted here.

**Comment by cas--** [comment]
I encountered the same issue. From the code it looks like `linkcheck_anchors_ignore` only matches against the just anchor and not the URL: https://github.com/sphinx-doc/sphinx/blob/master/sphinx/builders/linkcheck.py#L134

I think it might be intentional and the documentation is not correct. To ignore a site link with anchors use this regex in the **`linkcheck_ignore`** setting:

```python
linkcheck_ignore = [
    r'https://github.com/jonmiles/bootstrap-treeview/.*#'
]
```

**Comment by svenevs** [comment]
Oh I see, thank you for explaining that!  I will test that later today and if it works (I don't see why it wouldn't) I'll add a docs PR with an example to some anchor on the Sphinx readme itself :)

**Comment by tk0miya** [comment]
Fixed by #5185.
Thank you for reporting.
